Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -191,6 +191,83 @@ TEST_F(SerializerTest, EncodesStringModelToBytes) {
   }
 }
 
+TEST_F(SerializerTest, EncodesEmptyMapToBytes) {
+  FieldValue model = FieldValue::ObjectValue({});
+  // TEXT_FORMAT_PROTO: 'map_value: {}'
+  std::vector<uint8_t> bytes{0x32, 0x00};
+  ExpectRoundTrip(model, bytes, FieldValue::Type::Object);
+}
+
+TEST_F(SerializerTest, EncodesNestedObjectsToBytes) {
+  // As above, verify max int64_t value.
+  EXPECT_EQ(9223372036854775807, std::numeric_limits<int64_t>::max());
```

Comment:
1. This comment should explain the purpose of this check. Also, why not use `922...` number directly? In case you deliberately test for `max()` here, it would also be great to explain that in the comment.
2. This can probably be a `static_assert`. As far as I can tell, VS 2015 should have `numeric_limits<T>::max()` as `constexpr`.